### DISCUSSION TITLE
Handling URL components

### DISCUSSION THREAD

**Discussion by scollinson** [Kludex/uvicorn#1222] [Potential Issue]
While trying to create a basic HTTP proxy with starlette and uvicorn I've run into an interesting issue. I'm not sure if it's a problem with uvicorn or ASGI itself. 

The first thing I noticed was that uvicorn does not correctly handle the h11 request target when using it to create the scope `path` and `raw_path` variables. The h11 `Request.target` value is documented to have several forms as per https://datatracker.ietf.org/doc/html/rfc7230#section-5.3 and uvicorn only appears to handle the `origin-form` request target.

I updated `h11_impl.py` to use `urllib.parse.urlparse` on `event.target` to split out the host from the path, but noticed that the ASGI scope only stores the `path` and `query_string` and ignores path fragments and parameters. Is that intended? It seems like you'd not be able to reconstruct the complete query URL with the information supplied in the scope.

**Comment by euri10** [comment]
> but noticed that the ASGI scope only stores the path and query_string and ignores path fragments and parameters. Is that intended? 

I think that `query_string` in the scope will contain path fragments since it contains everything in the url after the `?`
